packages/angular/src/schematics/init/init.ts

```diff
@@ -0,0 +1,218 @@
+import {
+  convertNxGenerator,
+  generateFiles,
+  getProjects,
+  logger,
+  ProjectConfiguration,
+  readJsonFile,
+  Tree,
+  updateProjectConfiguration,
+} from '@nx/devkit';
+import { NgAddGeneratorSchema } from './schema';
+import * as path from 'path';
+import { fileURLToPath } from 'url';
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+import { EOL } from 'os';
+
+// Angular CLI specific configurtion
+export type ArchitectConfiguration = ProjectConfiguration['targets'];
+
+export async function ngAddGenerator(
+  tree: Tree,
+  options: NgAddGeneratorSchema
+) {
+  try {
+    const isAngularCli = tree.exists('angular.json') ? true : false;
+
+    const { build, serve, projectName, config, projectRoot } = isAngularCli
+      ? getElementsForAngularCli(tree, options)
+      : getElementsForNx(tree, options);
+
+    if (!build.configurations) {
+      build.configurations = {};
+    }
+
+    if (!serve.configurations) {
+      serve.configurations = {};
+    }
+
+    if (build.configurations['testronaut']) {
+      logger.info(
+        'Testronaut configuration already exists in build. Skipping configuration'
+      );
+      return;
+    }
+    if (serve.configurations['testronaut']) {
+      logger.info(
+        'Testronaut configuration already exists in serve. Skipping configuration'
+      );
+      return;
+    }
+
+    build.configurations['testronaut'] = {
+      optimization: false,
+      extractLicenses: false,
+      sourceMap: true,
+      browser: 'testronaut/main.ts',
+      index: 'testronaut/index.html',
+      tsConfig: 'testronaut/tsconfig.json',
+    };
```

Comment:
Not sure, if we should go after the name of the builder/generator. So I decided to use `main` only if it already present.